src/widgets/block.rs

```diff
@@ -917,4 +936,50 @@ mod tests {
             assert_buffer_eq!(buffer, Buffer::with_lines(vec![expected]));
         }
     }
+
+    #[test]
+    fn title_style_overrides_block_title_style() {
```

Comment:
I have a strong preference for keeping tests super simple. I found myself scrolling up to the test data and back down to the test itself to cross reference the data and behavior. We're testing 4 things that matter over 45 lines, which can be expressed as perhaps 4 tests each ~9 lines. The main benefit of splitting things this way is that the tests can help document the code. E.g. what about something like:

```rust
    #[test]
    fn title_content_style_renders_style() {
        for alignment in [Alignment::Left, Alignment::Center, Alignment::Right] {
            let mut buffer = Buffer::empty(Rect::new(0, 0, 4, 1));
            Block::default()
                .title_alignment(alignment)
                .title("test".yellow())
                .render(buffer.area, &mut buffer);
            let mut expected = Buffer::with_lines(vec!["test"]);
            expected.set_style(expected.area, Style::new().yellow());
            assert_buffer_eq!(buffer, expected);
        }
    }
```
etc.

Reply:
I was was testing basically 3 different cases, there are now separated in 3 tests and are easier to understand.